src/govuk/components/checkboxes/checkboxes.js

```diff
@@ -13,22 +13,18 @@ Checkboxes.prototype.init = function () {
   var $module = this.$module
   var $inputs = this.$inputs
 
-  /**
-  * Loop over all items with [data-controls]
-  * Check if they have a matching conditional reveal
-  * If they do, assign attributes.
-  **/
```

Comment:
Does this comment definitely need removing? It still feels useful to me - maybe excluding the line 
```
Loop over all items with [data-controls]
```

Or a different solution might be to rename
```
var target 
```
to
```
var conditionalReveal 
```

Reply:
I removed it as there's a lot of duplication between this comment and the comments within the callback function.

I've clarified the comments within the function, and I think they cover everything that was in this comment except the fact that it's a loop?

In terms of renaming the variable, the target element that's being associated is _probably_ (hopefully!) a conditional reveal, but I don't think we know that for sure at this point.

Reply:
As discussed earlier, we could add a general comment to indicate that the JavaScript here sets up the relationship of the `aria-controls` and conditional reveals.  

>  the target element that's being associated is probably (hopefully!) a conditional reveal, but I don't think we know that for sure at this point.

True. We could have a comment above that line to convey the _intention_ to extract the conditional reveal attribute (but it might not be necessary to add that if we add a comment elsewhere about the relationship).

Reply:
I've added JSDoc comments to each function, aiming to explain what the JS is doing.

@hannalaakso I'd value any thoughts or feedback – do you think this addresses your concerns?

Reply:
Yes the comments and clarified function names are super helpful 👍